tests/e2e/upgrade/upgrade_test.go

```diff
@@ -289,8 +297,60 @@ func TestUpgradePathHAModeMTLSEnabled(t *testing.T) {
 			upgradeOpts := common.TestOptions{
 				HAEnabled:   true,
 				MTLSEnabled: true,
-				// do not apply changes on upgrade, verify existing components.
-				ApplyComponentChanges: false,
+				// do not apply changes on upgrade, verify existing components and httpendpoints.
+				ApplyComponentChanges:    false,
+				ApplyHTTPEndpointChanges: false,
+				CheckResourceExists: map[common.Resource]bool{
+					common.CustomResourceDefs:  true,
+					common.ClusterRoles:        true,
+					common.ClusterRoleBindings: true,
+				},
+			}
+			tests := getTestsOnUpgrade(p, installOpts, upgradeOpts)
+
+			for _, tc := range tests {
+				t.Run(tc.Name, tc.Callable)
+			}
+		})
+	}
+}
+
+// HTTPEndpoint Dapr resource is a new type as of v1.11.
+// This test verifies install/upgrade functionality with this additional resource.
+func TestUpgradeWithHTTPEndpoint(t *testing.T) {
+	// Ensure a clean environment.
+	common.EnsureUninstall(false) // does not wait for pod deletion.
+	for _, p := range supportedUpgradePaths {
+		t.Run(fmt.Sprintf("setup v%s to v%s", p.previous.RuntimeVersion, p.next.RuntimeVersion), func(t *testing.T) {
+			t.Run("delete CRDs "+p.previous.RuntimeVersion, common.DeleteCRD(p.previous.CustomResourceDefs))
+			t.Run("delete CRDs "+p.next.RuntimeVersion, common.DeleteCRD(p.next.CustomResourceDefs))
+		})
+	}
+
+	for _, p := range supportedUpgradePaths {
+		// only check runtime versions that support HTTPEndpoint resource.
+		if p.next.RuntimeVersion != "v1.11.0-rc.4" {
```

Comment:
Please change to `if strings.Contains(p.next.RuntimeVersion, "1.11")` to support every 1.11.x version